Assets/SEE/Tools/Playername/PlayerName.cs

```diff
@@ -0,0 +1,123 @@
+﻿using UnityEngine;
+using TMPro;
+using Unity.Netcode;
+
+public class PlayerName : NetworkBehaviour
+{
+    /// <summary>
+    /// Reference to the TextMeshPro component to display the player's name.
+    /// </summary>
+    [SerializeField] private TMP_Text displayNameText;
+
+    /// <summary>
+    /// Variable to store the player's name.
+    /// </summary>
+    string playername;
+
+    private SEE.Net.Network networkConfig;
+
+    private void Start()
+    {
+        networkConfig = FindObjectOfType<SEE.Net.Network>();
+        if (networkConfig == null)
+        {
+            Debug.LogError("Network configuration not found");
+            return;
+        }
+        // Read the player's name when the script starts.
+        ReadPlayerName();
+    }
+
+    /// <summary>
+    /// Read the player's name from networkconfig.
+    /// </summary>
+    private void ReadPlayerName()
+    {
+        string playerName = networkConfig.Playername;
+
+        // If a player name is retrieved, store it.
+        if (playerName != null && playerName.Length != 0)
+        {
+            playername = playerName;
+        }
+        else
+        {
+            playername = "uknown Playername";
+        }
+    }
+
+    private void Update()
+    {
+        // If the NetworkObject is not yet spawned, exit early.
+        if (!IsSpawned)
+        {
+            return;
+        }
+
+        if (IsOwner)
+        {
+            DisplayLocalPlayername();
```

Comment:
Why does this need to be called every frame? Shouldn't it suffice to set the text once?

Reply:
If i put it into `Start` it doesn't work - not sure why.

Reply:
Please try to investigate why this is the case. Anything that we can move outside `Update()`, we should. By "it doesn't work", do you mean nothing happens, or some sort of error is thrown/appears in the console?

Reply:
i fixed this.